<a href="https://colab.research.google.com/github/NaghamZidiah/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I chose **Lane 3: Structured Content Archetype Clustering**.

I chose this lane because I want to discover different types of content pages based on their performance and characteristics instead of predicting a predefined label. By grouping similar pages into clusters, I can identify patterns such as high-performing pages, declining pages, or pages with engagement problems. These insights can help content teams better understand their content inventory and prioritize the most suitable actions for each group.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Decision:** Which content pages belong to different performance archetypes and should receive different optimization strategies?

**Action:** The content or SEO team can use these clusters to decide whether a page should be improved, monitored, protected, merged, or updated.

**Cost of a wrong recommendation:** If pages are assigned to the wrong cluster, the team may spend time improving pages that do not need attention or overlook pages that require immediate action. This can reduce the effectiveness of content optimization efforts and waste valuable resources.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/NaghamZidiah/FlyRank-ML-Internship.git

Cloning into 'FlyRank-ML-Internship'...
remote: Enumerating objects: 103, done.
remote: Counting objects: 100% (103/103), done.
remote: Compressing objects: 100% (74/74), done.
remote: Total 103 (delta 24), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (103/103), 1.84 MiB | 16.21 MiB/s, done.
Resolving deltas: 100% (24/24), done.


In [ ]:
%cd FlyRank-ML-Internship

/content/FlyRank-ML-Internship/FlyRank-ML-Internship


In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Number of content types:", df["content_type"].nunique())
print("Average content age (days):", round(df["content_age_days"].mean(), 2))

print("\nTrend distribution:")
print(df["trend_direction"].value_counts())

Number of pages: 30000
Number of content types: 3
Average content age (days): 256.17

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30,000 anonymized content pages.

The average content age is 256.17 days, which means the dataset includes pages with different maturity levels.

The trend distribution shows that 16,262 pages are currently labeled as down, indicating that content performance varies significantly and may contain different underlying patterns.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify observed patterns and directional differences between groups of content pages.

It can support decision-making by providing content archetypes and possible actions.

However, it cannot prove causal relationships, predict search engine algorithms, or guarantee that a specific action will improve performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.